content/ADR-213-quests-ui.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: adr
+adr: 213
+title: Quests UI
+date: 2023-04-17
+status: Review
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - guidota
+---
+
+## Abstract
+
+This ADR describes the decision-making process for implementing the user interface (UI) of the Quests System for a metaverse. We evaluated three potential implementation options and chose the one that strikes the best balance between flexibility and speed.
+
+## Context, Reach & Prioritization
+
+The Quests System's user interface is a crucial component of the metaverse experience, as it enables users to interact with quests and track their progress. As part of the UI design process, we need to determine if we want to use the SDK to create the UI components and in that case, how. The Explorer is the term used for our frontend implementation, which can take different forms. We need to decide how to balance the use of the SDK with the needs of the Explorer to create a user-friendly and consistent UI across all implementations.
+
+It is important to prioritize the decision-making process to ensure that we have enough time to design and implement the UI components before the launch of the Quests System. While using the SDK to create the UI components has its advantages, there are also limitations to what we can achieve with it. For instance, while we can use the SDK to implement components that support our use case, there may be gaps that we cannot address. Ultimately, we need to find a solution that allows us to create a visually appealing and functional UI that meets the needs of our users.
+
+## Solution Space Exploration
+
+There are three potential solutions for designing the Quests System UI:
+
+Using pure SDK components: In this approach, we would create the UI using only the components provided by the SDK. While this approach provides consistency across implementations, there may be some gaps that we would need to fill by implementing additional components.
+
+Using SDK to communicate the UI state to the Explorer: In this approach, we would use the SDK to store the UI state and communicate it to the Explorer. This would reduce the amount of communication required between the Explorer and the backend to retrieve information needed to render the UI. However, we would need to ensure that the SDK's capabilities are sufficient for our use case.
```

Comment:
```suggestion
Using SDK to communicate the UI state to the Explorer: In this approach, we would use the SDK to store the UI state and spread it to the Explorer. This would reduce the required communication between the Explorer and the backend to retrieve information needed to render the UI. However, we SHOULD ensure that the SDK's capabilities are sufficient for our use case.
```